back/app/services/input_ui_schema_generator_service.rb

```diff
@@ -123,4 +125,34 @@ def category_for(fields, category_id, translation_key)
       elements: fields.filter_map { |field| visit field }
     }
   end
+
+  def add_logic_rules(ui_schema, fields)
+    ordered_pages = fields.select(&:page?).sort_by(&:ordering)
+    ui_schema.tap do |schema|
+      schema.each_value do |locale_schema|
+        page_elements = locale_schema[:elements].select do |element|
+          element[:type] == 'Page'
+        end
+        page_elements.each do |page_element|
+          page_index = page_element.dig(:options, :id).split('_').last.to_i - 1
```

Comment:
This looks brittle. If we change the IDs of the page elements in the UI schema, then this will not work anymore. Isn't it enough to use `page_elements.each_with_index do |page_element, index|`?